### PR TITLE
macOS: Enable accessing a password reference without system prompt

### PR DESCRIPTION
**Pull request by roop**
In iOS, keychain sharing of password using access groups is enabled by default.

In macOS, it appears that we need to use the  `kSecUseDataProtectionKeychain` attribute to enable this behaviour. 

Current behaviour is that when the tunnel tries to access the tunnel password using the password reference, macOS shows a system prompt asking the user to authorize the access by typing in the user's system password. This PR enables accessing the tunnel password using the password reference without a system prompt.

As per the documentation on [Sharing Access to Keychain Items](https://developer.apple.com/documentation/security/keychain_services/keychain_items/sharing_access_to_keychain_items_among_a_collection_of_apps):

> This form of keychain item sharing applies to all iOS keychain items, and to macOS keychain items when you query with the kSecUseDataProtectionKeychain key, set the item’s kSecAttrSynchronizable attribute, or both.

### CODE REVIEW

**Comment by roop** [comment]
I tried to verify if Passpartout-iOS worked fine with this change, but I was unable to built it because of an unrelated issue that I couldn't resolve ('openssl/ssl.h' file not found).

I realize commit 4490f0c is related to this. From reading the code, I think this PR is still correct, but I apologise for not being able to verify this PR on top of that commit.

**Comment by keeshux** [comment]
After reviewing my code, I don't get why I dropped the app group constraint when locating a password reference, which is what made the signature eventually static. In hindsight I don't like it, still I would take the fix out of the context of this PR because it makes changes a bit misleading. Did you test if there's any difference with the addition/deletion of the app group parameter?

That said, I prefer fixing that in `master` and rebasing this PR afterwards. Do you agree?

**Comment by roop** [comment]
For iOS, the current code can be considered correct (with `password(for:, reference:, context:)` being a static function) because the code just searches all available access groups.

For macOS, however, this doesn't work because the use of `kSecUseDataProtectionKeychain` requires that we also specify the access group. The use of `kSecUseDataProtectionKeychain` (025f4ea) necessitates the passing of the access group (ca0e062), and consequently the  `password(for:, reference:, context:)` function becoming a member function. So these changes together do form a coherent fix in my mind.

Are you saying that making the dereferencing non-static (ca0e062) should preferably be fixed first in master, and then the addition of kSecUseDataProtectionKeychain key can be done (025f4ea) in this PR? I don't see why that's preferable, but I'm ok with any solution that has both these changes in master.

**Comment by keeshux** [comment]
> Are you saying that making the dereferencing non-static ([ca0e062](https://github.com/passepartoutvpn/tunnelkit/commit/ca0e062789ade5446e5fb31f6c9c489ac9e2038d)) should preferably be fixed first in master, and then the addition of kSecUseDataProtectionKeychain key can be done ([025f4ea](https://github.com/passepartoutvpn/tunnelkit/commit/025f4ea34d15d86c8515c21e478f58daec6aa897)) in this PR? I don't see why that's preferable, but I'm ok with any solution that has both these changes in master.

TL;DR I'm going to create another PR from ca0e062, then we'll rebase/merge this one.

I quoted the wrong commit which surely has caused confusion. I meant to say that after reverting the keychain method to non-static (ca0e062), this PR would be a one-liner (025f4ea). Which makes sense because given the subject of this PR, ca0e062 is not related, rather a partial fix/revert of 4490f0c.

Reason: this library is widely used in production and keychain is a dangerous place. If I'm to change something, I prefer it to be atomic and semantically separated. As is, the PR may look confusing in the future.

**Comment by roop** [comment]
Looks great, thanks. (And I sort of understand your point now.)
